File: src/main/java/me/nasukhov/TukitaLearner/bot/io/Input.java
package me.nasukhov.TukitaLearner.bot.io;

public record Input(String input, Channel channel, User sender) {
    public boolean isDirectCommand(String command) {
        return input.equals("/" + command) || input.startsWith("/" + command + "@");
    }

    @Override
    public String toString() {
        return input;
    }
}

File: src/main/kotlin/me/nasukhov/tukitalearner/bot/io/Input.kt
package me.nasukhov.tukitalearner.bot.io

data class Input(
    val raw: String,
    val channel: Channel,
    val sender: User,
) {
    fun isDirectCommand(command: String): Boolean = raw == "/$command" || raw.startsWith("/$command@")

    override fun toString(): String = raw
}
